### ISSUE TITLE
Crash on startup (OSX)

### ISSUE THREAD

**Issue by chrislloyd** [microsoft/vscode#90337]
- VSCode Version: 1.42
- OS Version: 10.15.3

Steps to Reproduce:

1. Open VSCode (either from Applications or `code` CLI)
2. Observe application crash (no window, just the Icon bouncing in the dock and disappearing)

Does this issue occur when all extensions are disabled?: Yes

```sh
$ code --disable-extensions --verbose
[main 2020-02-10T02:31:50.305Z] Error: ENOTDIR: not a directory, mkdir
[main 2020-02-10T02:31:50.309Z] Lifecycle#kill()
```

Roughly copying what seem to be relevant logs from OSX Console:

<img width="1249" alt="Screen Shot 2020-02-09 at 6 37 43 PM" src="https://user-images.githubusercontent.com/718/74117127-6a95af80-4b6b-11ea-8aa1-31caa5697661.png">

This doesn't occur with the Insider's build, however Insiders doesn't load any of my preferences. One unusual thing about my system is that my entire home directory is a git repo (though I'm unsure if that would affect the issue).

**Comment by bpasero** [comment]
@chrislloyd does it work if you use `code --user-data-dir <some folder>` from the command line?

**Comment by benediktberger** [comment]
No, that does open it! Thanks for the hint  :)

**Comment by bpasero** [comment]
@benediktberger then I think somehow you ended up having a file at a location where VSCode wants to create folders to persist data. Can you check the contents of ~/Library/Application Support/Code?

**Comment by benediktberger** [comment]
Thats the content:

> $ ls ~/Library/Application\ Support/Code/
1.19.2-shared.sock			1.23.1-shared.sock			1.28.0-shared.sock			Cookies-journal				in_progress_download_metadata_store
1.19.3-shared.sock			1.24.0-shared.sock			1.28.1-shared.sock			GPUCache				languagepacks.json
1.20.0-shared.sock			1.24.1-shared.sock			1.28.2-shared.sock			Local Storage				logs
1.20.1-shared.sock			1.25.0-shared.sock			Backups					Network Persistent State		machineid
1.21.0-shared.sock			1.25.1-shared.sock			Cache					Preferences				rapid_render.json
1.21.1-shared.sock			1.26.1-shared.sock			CachedData				User					storage.json
1.22.1-shared.sock			1.27.0-shared.sock			CachedExtensions			Workspaces				webrtc_event_logs
1.22.2-shared.sock			1.27.1-shared.sock			Code Cache				blob_storage
1.23.0-shared.sock			1.27.2-shared.sock			Cookies					clp

But I noticed something, may be related: As noted before, Code opens with no folder selected. When I try to open any folder, it just crashes again. May be related to the Apple "Open File" dialog - when I tried to navigate within the dialog, Code crashed.
